src/Icons/tests/Unit/IconRendererTest.php

```diff
@@ -0,0 +1,170 @@
+use PHPUnit\Framework\TestCase;
+use Symfony\UX\Icons\Exception\IconNotFoundException;
+use Symfony\UX\Icons\IconRegistryInterface;
+use Symfony\UX\Icons\IconRenderer;
+use Symfony\UX\Icons\Registry\InMemoryIconRegistry;
+use Symfony\UX\Icons\Svg\Icon;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ */
+class IconRendererTest extends TestCase
+{
+    public function testRenderIcon(): void
+    {
+        $registry = $this->createRegistry([
+            'user' => '<circle ',
+        ]);
+        $iconRenderer = new IconRenderer($registry);
+
+        $icon = $iconRenderer->renderIcon('user');
+
+        $this->assertStringContainsString('<svg', $icon);
+        $this->assertStringContainsString('<circle', $icon);
+    }
+
+    public function testRenderIconThrowsExceptionWhenIconNotFound(): void
+    {
+        $registry = $this->createRegistry([]);
+        $iconRenderer = new IconRenderer($registry);
+
+        $this->expectException(IconNotFoundException::class);
+
+        $iconRenderer->renderIcon('foo');
+    }
+
+    public function testRenderIconThrowsExceptionWhenAttributesAreInvalid(): void
+    {
+        $registry = $this->createRegistry(['foo' => '<path d="M0 0L12 12"/>']);
+        $iconRenderer = new IconRenderer($registry);
+
+        $this->expectException(InvalidArgumentException::class);
+
+        $iconRenderer->renderIcon('foo', [1, 2, null]);
+    }
+
+    public function testRenderIconWithAttributes(): void
+    {
+        $registry = $this->createRegistry([
+            'foo' => '<path d="M0 0L12 12"/>',
+        ]);
+        $iconRenderer = new IconRenderer($registry);
+        $attributes = ['viewBox' => '0 0 24 24', 'class' => 'icon', 'id' => 'FooBar'];
+
+        $svg = $iconRenderer->renderIcon('foo', $attributes);
+
+        $this->assertSame('<svg viewBox="0 0 24 24" class="icon" id="FooBar"><path d="M0 0L12 12"/></svg>', $svg);
+    }
+
+    public function testRenderIconWithDefaultAttributes(): void
+    {
+        $registry = $this->createRegistry([
+            'foo' => '<path d="M0 0L12 12"/>',
+        ]);
+        $iconRenderer = new IconRenderer($registry, ['viewBox' => '0 0 24 24', 'class' => 'icon']);
+
+        $svg = $iconRenderer->renderIcon('foo');
+
+        $this->assertSame('<svg viewBox="0 0 24 24" class="icon"><path d="M0 0L12 12"/></svg>', $svg);
+    }
+
+    /**
+     * @dataProvider provideAttributesWithDefaultAttributesCases
+     */
+    public function testRenderIconWithAttributesAndDefaultAttributes($iconAttrs, $defaultAttrs, $renderAttr, $expectedTag): void
+    {
+        $registry = $this->createRegistry([
+            'foo' => ['', $iconAttrs],
+        ]);
+        $iconRenderer = new IconRenderer($registry, $defaultAttrs);
+
+        $svg = $iconRenderer->renderIcon('foo', $renderAttr);
+        $this->assertStringStartsWith($expectedTag, $svg);
+    }
+
+    /**
+     * @return array [iconAttributes, defaultAttributes, renderAttributes, expectedTag]
```

Comment:
Is this valid phpdoc syntax?

Reply:
A iterable/generator of array but yes. I'll delete all as no one will add tests